sui-execution/v1/sui-adapter/src/temporary_store.rs

```diff
@@ -93,8 +93,12 @@ impl<'backing> TemporaryStore<'backing> {
     }
 
     pub fn update_object_version_and_prev_tx(&mut self) {
-        self.execution_results
-            .update_version_and_previous_tx(self.lamport_timestamp, self.tx_digest);
+        self.execution_results.update_version_and_previous_tx(
+            self.lamport_timestamp,
+            self.tx_digest,
+            &self.input_objects,
+            self.protocol_config.reshare_at_same_initial_version(),
```

Comment:
for the older execution versions, should we assert that this is false? or pass in false explicitly?

Reply:
We can pass in false explicitly (was on the fence on it)